### PR TITLE
Change deprecate linting errors to warn

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

This is meant to improve the developer experience. We have an ESLint rule which runs on a `precommit` hook and checks any changed files to see if they are using any "deprecated" Design System components. In this case, a deprecated component is any React component which has a Web Component replacement. With its current configuration, the linter will throw errors and block a commit if a deprecated import is found in a changed file, event if the changes have nothing to do with the component.

This PR changes the errors to a warning and also updates the npm script so that the warnings are shown in the cli.

More context in [a slack thread](https://dsva.slack.com/archives/CBU0KDSB1/p1622565642342700).

## Testing done

Testing with local changes to trigger the linter.

## Screenshots

image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by erikphansen — COMMENTED** [review]
This doesn't seem to give any indication within VSCode, which would probably be the most effective way to let devs know they should start using `va-alert` instead of `AlertBox` without getting in their way. Also I think a lot of folks will miss the pre-commit hook output. If you use a Git GUI you'll likely never even see the warning.

**Review by jhonny218 — APPROVED** [review]

**Comment by bkjohnson** [comment]
@erikphansen I'm having some problems with linting locally, but do you think it would be better to get rid of `changed.eslintrc.js` entirely and move the `deprecate` rule into the main ESLint config as a warning?

**Comment by jbalboni** [comment]
@bkjohnson I like that. If it's a warning, it'd be nice for it to show up in more places.

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "lint:js": "eslint --quiet --ext .js --ext .jsx .",
     "lint:circle": "eslint --ext .js --ext .jsx -c circle.eslint.json .",
     "lint:js:fix": "eslint --fix --quiet --ext .js --ext .jsx .",
-    "lint:js:changed": "LIST=`git diff-index --diff-filter=d --name-only --cached HEAD | grep \"\\.js[x]\\{0,1\\}$\"`; if [ \"$LIST\" ]; then eslint --quiet --config changed.eslintrc.js $LIST; fi",
+    "lint:js:changed": "LIST=`git diff-index --diff-filter=d --name-only --cached HEAD | grep \"\\.js[x]\\{0,1\\}$\"`; if [ \"$LIST\" ]; then eslint --quiet $LIST; fi",
```
Comment (bkjohnson) [inline package.json:31]: This is to get the `lint:js:changed` script back to where it was before https://github.com/department-of-veterans-affairs/vets-website/pull/16270

File: changed.eslintrc.js [hunk 2]
```diff
@@ -1,21 +0,0 @@
-const eslintConfig = require('./.eslintrc');
-
-// Add the config for preventing deprecated component-library imports
-eslintConfig.plugins = [...eslintConfig.plugins, 'deprecate'];
-eslintConfig.rules = {
-  ...eslintConfig.rules,
-  'deprecate/import': [
-    'error',
```
Comment (bkjohnson) [inline changed.eslintrc.js:8]: This rule is moved into the main ESLint config with the exception that `error` is changed to `warn`.

**Review by erikphansen — APPROVED** [review]

**Pull request merged**